include/utils/CreateDeviceExpression.h

```diff
@@ -0,0 +1,27 @@
+#ifndef CREATEDEVICEEXPRESSION_H
+#define CREATEDEVICEEXPRESSION_H
+
+#include <type_traits>
+
+#include <KokkosInterface.h>
+
+namespace sierra {
+namespace nalu {
+
+template <typename T>
+inline 
+T* create_device_expression(const T & rhs)
+{
+  const std::string debuggingName(typeid(T).name());
+  T* t = kokkos_malloc_on_device<T>(debuggingName);
+  // Bring rhs into local scope for capture to device.
+  const T RHS(rhs);
+  kokkos_parallel_for(debuggingName, 1, [&] (const int i) {
```

Comment:
if rhs was passed in by value instead of reference, then would the local RHS be un-necessary?

Reply:
Overall looks great, I like the unit test.

Reply:
Yes, passing by value would remove the need for a local RHS and I agree it would make the code a bit cleaner.

Reply:
Wait, darn, should have read my own comment.  I don't think the cuda compiler will correctly bring a passed parameter into local scope and mirror it to the device.  It should but we have had trouble with this.  We could pass by value and then set a reference to it to bring it into local scope.  Could also be that newer versions of the compiler fix this.

Reply:
That's surprising. It seems like passing something by value through the argument list should be the same as constructing the local from a reference. Both necessarily involve copying the class data from the host to the device. In any case, this isn't a big deal. Perhaps we could query Christian Trott just to hear his insight on this issue.